### DISCUSSION TITLE
Trigger function with Controller

### DISCUSSION THREAD

**Discussion by timheerwagen** [react-hook-form/react-hook-form#9858] [Ideas]
How about having the trigger function output in the controller as well, so that you can validate the value used by the controller individually?

Currently I need to pass the trigger function from useForm to the child component. But it would be more convenient to get it directly from a controller.

**Comment by Moshyfawn** [comment]
I don't think we want to expand the API surface like that just now.

At this time, you can either pass it as a prop or by `FormProvider` and get it with `useFormContext`
